indra/llcharacter/llik.cpp

```diff
@@ -271,35 +275,26 @@ size_t LLIK::Constraint::generateHash() const
     return seed;
 }
 
-
+// returns 'true' if joint was adjusted
 bool LLIK::Constraint::enforce(Joint& joint) const
 {
+    // This enforce() method of the base-Constraint provides no back-pressure
+    // on joint's parent when adjusting joint's local-frame transform.
+    // Override this method as necessary to provide back-pressure for select
+    // derived Constraints.
     const LLQuaternion& local_rot = joint.getLocalRot();
     LLQuaternion adjusted_local_rot = computeAdjustedLocalRot(local_rot);
     if (! LLQuaternion::almost_equal(adjusted_local_rot, local_rot))
     {
+        // Note: we update joint's local-frame rot, but not its world-frame rot
+        // -- that responsibility belongs to external code.
         joint.setLocalRot(adjusted_local_rot);
+        joint.applyLocalRot();
         return true;
     }
```

Comment:
`joint.applyLocalRot(`) used to be called explicitly after a constraint had been enforced, so I moved it inside this context.